### PR TITLE
add create department and add create position pages

### PR DESCRIPTION
**Pull request by MariyaSin**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/626

### CODE REVIEW

File: src/GUI/GUI.csproj [hunk 1]
```diff
@@ -7,6 +7,18 @@
     <RootNamespace>LT.DigitalOffice.GUI</RootNamespace>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (BelehovEgor) [inline src/GUI/GUI.csproj]: Do we need it?

File: src/GUI/Pages/Admin/Company/CreateDepartment.razor [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    <p>@message</p>
+    <button class="btn btn-primary" @onclick="@Post">Create department</button>
+</div>
+
+@code {
+    private NewDepartmentRequest departmentData = new();
+    private BaseDepartmentInfo departmentInfo = new();
+    public string message;
+    private ElementReference descriptionInput;
+    private ElementReference directorSelect;
```
Comment (KostinP) [inline src/GUI/Pages/Admin/Company/CreateDepartment.razor]: Useless?

**Review by KostinP — APPROVED** [review]

File: src/GUI/Pages/Admin/Company/CreateDepartment.razor (hunk 2 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/Company/CreateDepartment.razor]: fixed

**Review by TIove — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

File: src/GUI/Pages/Admin/Company/CreateDepartment.razor [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        </div>
+        <div class="create-department_actions-container">
+            <p class="info_container">@message</p>
+            <button class="btn btn-primary" type="submit">@Resources.CreateDepartment</button>
+        </div>
+    </div>
+</EditForm>
+
+    @code {
+        private NewDepartmentRequest departmentData = new();
```
Comment (djino97) [inline src/GUI/Pages/Admin/Company/CreateDepartment.razor]: all private variables begin write with underscore

File: src/GUI/Pages/Admin/Company/CreatePosition.razor [hunk 4]
```diff
@@ -0,0 +1,47 @@
+﻿@using System.Threading.Tasks;
+@using LT.DigitalOffice.GUI.Services.Interfaces;
+@using LT.DigitalOffice.GUI.Services.ApiClients.CompanyService;
+@using LT.DigitalOffice.GUI.Properties;
+
+@inject ICompanyService companyService
+
+<EditForm Model=@positionData OnValidSubmit="@HandleValidSubmit">
```
Comment (djino97) [inline src/GUI/Pages/Admin/Company/CreatePosition.razor]: Model="@positionData"

File: src/GUI/Pages/Admin/Company/CreatePosition.razor [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    <DataAnnotationsValidator />
+    <div class="create-position_container">
+        <div class="item-groop">
+            <label for="create-department_name">@Resources.PositionName*</label>
```
Comment (djino97) [inline src/GUI/Pages/Admin/Company/CreatePosition.razor]: use `<font>` for *

File: src/GUI/Services/CompanyService.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+using LT.DigitalOffice.GUI.Helpers;
+using System;
+
+namespace LT.DigitalOffice.GUI.Services
+{
+    public class CompanyService : ICompanyService
+    {
+        private readonly ISessionStorageService _storage;
+        private readonly CompanyServiceClient _client = new CompanyServiceClient(new System.Net.Http.HttpClient());
+        private string _token;
```
Comment (djino97) [inline src/GUI/Services/CompanyService.cs:15]: add empty line

File: src/GUI/Services/CompanyService.cs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            }
+        }
+
+        public async Task<DepartmentsResponse> GetDepartments()
+        {
+            try
+            {
+                _token = await _storage.GetItemAsync<string>(Consts.Token);
+                return await _client.GetDepartmentsAsync(_token);
+
```
Comment (djino97) [inline src/GUI/Services/CompanyService.cs]: remove empty line

File: src/GUI/Services/CompanyService.cs [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            }
+            catch (ApiException<ErrorResponse> ex)
```
Comment (djino97) [inline src/GUI/Services/CompanyService.cs:70]: Why use ApiException instead Exception?

File: src/GUI/Services/CompanyService.cs [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                //to do when spec changed
+                return null;
+            }
+        }
+
+        public async Task<ICollection<PositionResponse>> GetPositions()
+        {
+            try
+            {
+                _token = await _storage.GetItemAsync<string>(Consts.Token);
```
Comment (djino97) [inline src/GUI/Services/CompanyService.cs:81]: add empty line

File: src/GUI/Services/Interfaces/ICompanyService.cs [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+{
+    public interface ICompanyService
+    {
+        public Task<string> CreateDepartment(NewDepartmentRequest request);
+
+        public Task<string> CreatePosition(CreatePositionRequest request);
+
+        public Task<DepartmentsResponse> GetDepartments();
+
+        public Task<ICollection<PositionResponse>> GetPositions();
```
Comment (djino97) [inline src/GUI/Services/Interfaces/ICompanyService.cs]: remove public

File: src/GUI/Services/UserService.cs [hunk 11]
```diff
@@ -10,6 +10,7 @@ namespace LT.DigitalOffice.GUI.Services
     public class UserService : IUserService
     {
         private readonly ISessionStorageService _sessionStorage;
+        private readonly UserServiceClient _client = new UserServiceClient(new System.Net.Http.HttpClient());
```
Comment (djino97) [inline src/GUI/Services/UserService.cs]: create instance in constructor class

File: src/GUI/Pages/Admin/Company/CreatePosition.razor (hunk 4 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/Company/CreatePosition.razor]: done

File: src/GUI/Services/CompanyService.cs (hunk 6 above)
Comment (MariyaSin) [reply src/GUI/Services/CompanyService.cs:15]: done

File: src/GUI/Services/CompanyService.cs (hunk 7 above)
Comment (MariyaSin) [reply src/GUI/Services/CompanyService.cs]: done

File: src/GUI/Services/CompanyService.cs (hunk 8 above)
Comment (MariyaSin) [reply src/GUI/Services/CompanyService.cs:70]: replied to private messages

File: src/GUI/Services/CompanyService.cs (hunk 9 above)
Comment (MariyaSin) [reply src/GUI/Services/CompanyService.cs:81]: done

File: src/GUI/Services/Interfaces/ICompanyService.cs (hunk 10 above)
Comment (MariyaSin) [reply src/GUI/Services/Interfaces/ICompanyService.cs]: done

File: src/GUI/Pages/Admin/Company/CreateDepartment.razor (hunk 3 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/Company/CreateDepartment.razor]: done

File: src/GUI/Services/UserService.cs (hunk 11 above)
Comment (MariyaSin) [reply src/GUI/Services/UserService.cs]: done

File: src/GUI/GUI.csproj (hunk 1 above)
Comment (MariyaSin) [reply src/GUI/GUI.csproj]: removed

File: src/GUI/Pages/Admin/Company/CreatePosition.razor (hunk 5 above)
Comment (MariyaSin) [reply src/GUI/Pages/Admin/Company/CreatePosition.razor]: done

**Review by djino97 — APPROVED** [review]

**Pull request merged**